### PR TITLE
AADS, on AutoML, when I use predict.py for Image Classification, It works for asset without labels

### PR DESCRIPTION
**Pull request by FlorianLeRoyKili**

### CODE REVIEW

**Review by PierreLeveau — CHANGES_REQUESTED** [review]
can you add the script used to create image dataset for e2e tests?

File: kiliautoml/models/_pytorchvision_image_classification.py [hunk 1]
```diff
@@ -194,27 +211,56 @@ def find_errors(
...
+            label = get_label(asset, label_merge_strategy)
+            if (label is None) or (self.job_name not in label["jsonResponse"]):
+                asset_id = asset["id"]
+                warnings.warn(f"${asset_id}: No annotation for job ${self.job_name}")
+                return 0.0
+            else:
+                labels.append(label["jsonResponse"][self.job_name]["categories"][0]["name"])
+
         kf = StratifiedKFold(n_splits=cv_n_folds, shuffle=True, random_state=42)
+        pyx = np.empty((len(labels), len(self.class_name_to_idx)))
```
Comment (crsegerie) [inline kiliautoml/models/_pytorchvision_image_classification.py:228]: rename to "probability_matrix"

File: kiliautoml/models/_pytorchvision_image_classification.py [hunk 2]
```diff
@@ -194,27 +211,56 @@ def find_errors(
...
+        images = download_project_images(
+            api_key=api_key, assets=assets, output_folder=self.data_dir
+        )
+        labels = []
+        for asset in assets:
+            label = get_label(asset, label_merge_strategy)
+            if (label is None) or (self.job_name not in label["jsonResponse"]):
+                asset_id = asset["id"]
+                warnings.warn(f"${asset_id}: No annotation for job ${self.job_name}")
+                return 0.0
```
Comment (crsegerie) [inline kiliautoml/models/_pytorchvision_image_classification.py:223]: why 0.0?

File: kiliautoml/models/_pytorchvision_image_classification.py [hunk 3]
```diff
@@ -194,27 +211,56 @@ def find_errors(
...
-                original_image_datasets,
-                cv_fold,
-                cv_train_idx,
-                cv_holdout_idx,
+            splits = {}
+            splits["train"], splits["val"] = train_test_split(
+                cv_train_idx, test_size=0.2, random_state=42
             )
+            image_datasets = {
+                x: ClassificationTrainDataset(
```
Comment (crsegerie) [inline kiliautoml/models/_pytorchvision_image_classification.py:238]: does it make sense to rename to  ClassificationDataset ?

File: kiliautoml/utils/pytorchvision/trainer.py [hunk 4]
```diff
@@ -25,19 +25,20 @@ def train_model_pytorch(
...
+    dataset_sizes = {x: len(dataloaders[x].dataset) for x in ["train", "val"]}
 
     # Decay LR by a factor of 0.1 every 7 epochs
     scheduler = lr_scheduler.StepLR(optimizer, step_size=7, gamma=0.1)
 
     best_model_wts = copy.deepcopy(model.state_dict())
     best_acc = 0.0
-
+    best_loss = float("inf")
     for epoch in range(epochs):
```
Comment (crsegerie) [inline kiliautoml/utils/pytorchvision/trainer.py:42]: add tqdm here
from tqdm.autonotebook import tqdm

Comment (crsegerie) [reply kiliautoml/utils/pytorchvision/trainer.py:42]: trange(epochs, desc='Training')

File: kiliautoml/models/_pytorchvision_image_classification.py (hunk 2 above)
Comment (FlorianLeRoyKili) [reply kiliautoml/models/_pytorchvision_image_classification.py:223]: It was because train need to return a float. But with find_errors, you are right it must return a list of assets so I changed it to []

File: kiliautoml/models/_pytorchvision_image_classification.py (hunk 3 above)
Comment (FlorianLeRoyKili) [reply kiliautoml/models/_pytorchvision_image_classification.py:238]: It is to differentiate between ClassificationTrainDataset and ClassificationPredictDataset where one store labels, the other not. But I am open to a renaiming

File: kiliautoml/utils/pytorchvision/trainer.py (hunk 4 above)
Comment (FlorianLeRoyKili) [reply kiliautoml/utils/pytorchvision/trainer.py:42]: OK, but do I keep the             
```
print(f"Epoch {epoch + 1}/{epochs}")
print("-" * 10)
```
then ?

Comment (FlorianLeRoyKili) [reply kiliautoml/utils/pytorchvision/trainer.py]: Ok, after tries, I put 
```
    for _ in trange(epochs, desc="Training - Epoch"):
        if verbose >= 2:
            print("-" * 10)

```

**Review by PierreLeveau — APPROVED** [review]
